log-collector/src/main/scala/io/phdata/pulse/logcollector/LogCollector.scala

```diff
@@ -53,24 +56,48 @@ object LogCollector extends LazyLogging {
 
     val routes = new LogCollectorRoutes(solrService)
 
-    // Starts Http Service
-    def start(port: Int): Future[Unit] =
-      Http().bindAndHandle(routes.routes, "0.0.0.0", port = port)(materializer) map { binding =>
+    def serve(port: Int): Unit = {
```

Comment:
This looks like a function in a function, this should be defined at the class level.